obstore/python/obstore/fsspec.py

```diff
@@ -177,15 +224,58 @@ async def _ls(self, path, detail=True, **kwargs):
         else:
             return sorted([object["path"] for object in objects] + prefs)
 
-    def _open(self, path, mode="rb", **kwargs):
+    def _open(self, path: str, mode="rb", **kwargs):
         """Return raw bytes-mode file-like from the file-system"""
-        return BufferedFileSimple(self, path, mode, **kwargs)
+        assert mode in (
+            "rb",
+            "wb",
+        ), f"Only 'rb' and 'wb' mode is currently supported, got: {mode}"
```

Comment:
```suggestion
        if mode not in {"rb", "wb"}:
            raise ValueError(f"Only 'rb' and 'wb' mode is currently supported, got: {mode}")
```
I know @kylebarron said "assert", but I think this is clearer. It is unusual to see `assert` outside of tests.